obelisk/python/obelisk_py/core/utils/launch_utils.py

```diff
@@ -340,6 +340,84 @@ def _single_viz_node_launch_actions(settings: Dict, suffix: Optional[int] = None
     return launch_actions
 
 
+def get_launch_actions_from_joystick_settings(settings: Dict, global_state_node: LifecycleNode) -> List[LifecycleNode]:
+    """Gets and configures all the launch actions related to joystick given the settings from the yaml."""
+    launch_actions = []
+    if settings["on"]:
+        # TODO: I think there is a better way to handle these defaults, i.e., not even passing the parameter if nothing
+        # is supplied.
+        if "device_id" in settings:
+            device_id = settings["device_id"]
+        else:
+            device_id = 0
```

Comment:
to shorten this code you can do
```
device_id = settings["device_id"] if "device_id" in settings else 0
```
even better, we should define a utility function for this:
```
def parse_joystick_params(settings: Dict, param_names: List[String], default_values: List[Any]):
    params = {}
    for param_name, default_value in zip(param_names, default_values):
        params[param_name] = settings[param_name] if param_name in settings else default_value
    return params

# use it like
Node(
    # other args
    parameters[parse_joystick_params(
            settings,
            ["device_id", "device_name", "deadzone", ...],  # TODO: fill these in
            [0, "", 0.05, ...],  # TODO: fill these in
        )
    ]
)
```